app/src/main/java/com/nervousfish/nervousfish/data_objects/RSAKey.java

```diff
@@ -48,36 +52,12 @@ public RSAKey(final Map<String, String> map) throws IllegalArgumentException {
         }
     }
 
-    /**
-     * Get the exponent of an {@link RSAKey}.
-     *
-     * @param key An {@link RSAKey}.
-     * @return The value for the exponent attribute of the {@code key}.
-     * @throws Exception If the provided {@link IKey} is not a {@link RSAKey}.
-     */
-    public static String getExponent(final IKey key) throws IllegalArgumentException {
-        final String type = key.getType();
-        if (!type.equals(ConstantKeywords.RSA_KEY)) {
-            throw new IllegalArgumentException();
-        }
-
-        return ((RSAKey) key).exponent;
+    public String getExponent() {
```

Comment:
I'm not so sure. When you get the exponent / modulus from a key and you work the way as you proposed, then you have no idea if the exponent / modulus you got are valid. You'll have to write code around your calls to the getters to check if they are valid (so from an RSA key) or invalid (from, for example, a SimpleKey). That's very error prone and a design smell.

I would very much like to propose a better way to solve this, maybe with a Visitor pattern, but because the getExponent() and getModulus() methods are not used anywhere I cannot. I think it's best to delete them for now, because they are not used, and when they are needed in the future, that we'll then find an elegant solution that suits that situation.